File: pgen.py
import random
import string

def generate_password(length=12, use_uppercase=True, use_numbers=True, use_special_chars=True):
    # Define character sets
    lowercase = string.ascii_lowercase
    uppercase = string.ascii_uppercase
    digits = string.digits
    special_chars = string.punctuation

    # Start with lowercase letters
    characters = lowercase

    # Add uppercase letters if needed
    if use_uppercase:
        characters += uppercase

    # Add digits if needed
    if use_numbers:
        characters += digits

    # Add special characters if needed
    if use_special_chars:
        characters += special_chars

        # Generate the password by randomly selecting characters
    password = ''.join(random.choice(characters) for i in range(length))
    
    return password

File: pgen.js
document.getElementById("generate-btn").addEventListener("click", function () {
    const length = document.getElementById("length").value;
    const includeUppercase = document.getElementById("uppercase").checked;
    const includeNumbers = document.getElementById("numbers").checked;
    const includeSpecialChars = document.getElementById("special").checked;

    const password = generatePassword(length, includeUppercase, includeNumbers, includeSpecialChars);
    document.getElementById("password").textContent = password;
});

function generatePassword(length, includeUppercase, includeNumbers, includeSpecialChars) {
    const lowercase = "abcdefghijklmnopqrstuvwxyz";
    const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    const digits = "0123456789";

    const specialChars = "!@#$%^&*()-_=+[]{}|;:,.<>?/~";
    
    let characters = lowercase;
    
    if (includeUppercase) characters += uppercase;
    if (includeNumbers) characters += digits;
    if (includeSpecialChars) characters += specialChars;

    let password = "";
    for (let i = 0; i < length; i++) {
        const randomIndex = Math.floor(Math.random() * characters.length);
        password += characters[randomIndex];
    }
    
    return password;
}